README.md

```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
+  public/ - The public application API
+  cache/ - The cache interface that can be implemented to provide persistence cache storage of credentials
+```
+
 Acquiring tokens with MSAL Go follows this general three step pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
+
 1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
-```go
-publicClientApp, err := msal.CreatePublicClientApplication("your_client_id",
-    "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here")
-```
+   * Initializing a public client:
 
-2. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
-```go
-accounts := publicClientApp.GetAccounts()
-// Assuming the user wanted the first account
-userAccount := accounts[0]
-scopes := []string{"your_scope"}
-silentParams := msal.CreateAcquireTokenSilentParametersWithAccount(scopes, userAccount)
-result, err := publicClientApp.AcquireTokenSilent(silentParams)
-```
+    ```go
+    app, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+   * Initializing a confidential client:
+
+    ```go
+    // Initializing the client credential
+    cred, err := confidential.NewCredFromSecret("client_secret")
+    if err != nil {
+        return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+    }
+    app, err := confidential.New("client_id", cred, confidential.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+1. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
+
+    ```go
+    var userAccount public.Account
+    accounts := app.Accounts()
+    if len(accounts) > 0 {
+        // Assuming the user wanted the first account
+        userAccount = accounts[0]
+        // found a cached account, now see if an applicable token has been cached
+        authResult, err := app.AcquireTokenSilent(context.Background(), []string{"your_scope"}, public.WithSilentAccount(userAccount))
+    }
+    ```
 
-3. If there is no suitable token in the cache, or you choose to skip this step, now we can send a request to AAD to obtain a token. 
-```go
-if err != nil {
-    // Based on the flow, you create the type of parameters
-    tokenParams := msal.CreateAcquireTokenxxxParameters(scopes, ...)
-    result, err := publicClientApp.AcquireTokenByxxx(tokenParams)
+1. If there is no suitable token in the cache, or you choose to skip this step, now we can send a request to AAD to obtain a token.
+
+    ```go
+    result, err := app.AcquireTokenByOneofTheActualMethods([]string{"your_scope"}, ...(other parameters depending on the function) )
```

Comment:
>ByOneofTheActualMethods [](start = 35, length = 23)

should it be in quotes?

Reply:
Although not entirely runnable code, it still is a sample code snippet so as to distinguish it from the text. I think we should have the quotes.

Reply:
I meant: "ByOneOfTheActualMethods" is not quoted afaiks? I think it should be

---
In reply to: [562169548](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146#discussion_r562169548) [](ancestors = 562169548)

Reply:
Got it, changed in the latest [commit](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146/commits/4179b9ba833b42c58ed822ee1f09c2138f1175a6)